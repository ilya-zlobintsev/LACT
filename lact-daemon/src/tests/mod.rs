use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
