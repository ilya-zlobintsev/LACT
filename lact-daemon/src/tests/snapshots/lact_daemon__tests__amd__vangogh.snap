---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": null,
  "info": {
    "driver": "amdgpu",
    "link_info": {
      "current_speed": "8.0 GT/s PCIe",
      "current_width": "16"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "VanGogh [AMD Custom GPU 0405]",
        "model_id": "163F",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model_id": "0123",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      }
    },
    "vbios_version": "113-AMDAerithJ-005",
    "vulkan_info": {
      "api_version": "",
      "device_name": "",
      "driver": {
        "version": 0
      },
      "enabled_layers": [],
      "extensions": {},
      "features": {}
    }
  },
  "pci_info": {
    "device_pci_info": {
      "model": "VanGogh [AMD Custom GPU 0405]",
      "model_id": "163F",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model_id": "0123",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    }
  },
  "power_profile_modes": null,
  "power_states": {
    "core": [
      {
        "enabled": true,
        "index": 0,
        "value": 200
      },
      {
        "enabled": true,
        "index": 1,
        "value": 1100
      },
      {
        "enabled": true,
        "index": 2,
        "value": 1600
      }
    ],
    "vram": [
      {
        "enabled": true,
        "index": 0,
        "value": 400
      },
      {
        "enabled": true,
        "index": 1,
        "value": 600
      },
      {
        "enabled": true,
        "index": 2,
        "value": 687
      },
      {
        "enabled": true,
        "index": 3,
        "value": 687
      }
    ]
  },
  "stats": {
    "busy_percent": 0,
    "clockspeed": {
      "gpu_clockspeed": 200
    },
    "core_power_state": 0,
    "fan": {
      "control_enabled": false,
      "pmfw_info": {}
    },
    "power": {
      "average": 3.218,
      "cap_current": 15.0,
      "cap_default": 15.0,
      "cap_max": 29.0,
      "cap_min": 0.0,
      "current": 4.057
    },
    "temps": {
      "edge": {
        "crit": null,
        "crit_hyst": null,
        "current": 56.0
      }
    },
    "voltage": {
      "gpu": 15,
      "northbridge": 669
    },
    "vram": {}
  }
}
